client/verta/verta/registry/_verta_model_base.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+
+import abc
+
+from verta.external import six
+
+
+@six.add_metaclass(abc.ABCMeta)
+class VertaModelBase(object):
+    """Abstract base class for Verta Standard Models.
+
+    .. note::
+
+        ``__init__()`` and :meth:`predict` **must** be implemented by
+        subclasses. Any other methods are optional.
+
+    Parameters
+    ----------
+    artifacts : dict of str to str
+        A mapping of artifact keys to filepaths. This will be provided to the
+        deployed model based on artifact keys specified through
+        :meth:`RegisteredModelVersion.log_model()
+        <verta.registry.entities.RegisteredModelVersion.log_model>`.
+
+    Examples
+    --------
+    .. code-block:: python
+
+        import pickle
+        import numpy as np
+        from verta.registry import VertaModelBase
+
+        class Model(VertaModelBase):
+            def __init__(self, artifacts):
+                with open(artifacts["np_matrix"], "rb") as f:
+                    self._transform = pickle.load(f)
+
+            def predict(self, input):
+                input = np.array(input)
+
+                return np.matmul(input, self._transform)
+
+        # iterate locally
+        model = Model(
+            artifacts=model_ver.fetch_artifacts(["np_matrix"]),
+        )
+
+        # persist to model version
+        model_ver.log_model(Model, artifacts=["np_matrix"])
+
+    """
+
+    @abc.abstractmethod
+    def __init__(self, artifacts):
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def predict(self, input):
+        """Produce an output from `input`.
+
+        This method is called when requests are made against a Verta endpoint.
+
+        .. note::
+
+            This method should expect to recieve and return values that are
```

Comment:
Should this expect to do the deserialization itself? what is compatible? What's not?

Reply:
> Should this expect to do the deserialization itself?

It should expect that the value it gets as `input` is a Python primitive, and it should return Python primitives. If it wants to handle a NumPy array, `predict()`'s body will need to manually cast it (`input = np.array(input)`). Does that make sense? Would you have wording recommendation that might make this clearer?

---
> what is compatible? What's not?

I was hoping that the
> JSON-compatible (most Python built-in types)

would explain this, but maybe it's not as clear as I expected. The answer is `tuple`, `list`, `dict`, `int`, `float`, `bool`, `str`, `None`. I can list these out explicitly.